Zend/zend_compile.c

```diff
@@ -5863,6 +5865,49 @@ void zend_compile_params(zend_ast *ast, zend_ast *return_type_ast, uint32_t fall
 			opline->op2.num = type_ast ?
 				ZEND_TYPE_FULL_MASK(arg_info->type) : MAY_BE_ANY;
 		}
+
+		if (visibility) {
+			zend_op_array *op_array = CG(active_op_array);
+			zend_class_entry *scope = op_array->scope;
+			zend_bool is_ctor =
+				scope && zend_string_equals_literal_ci(op_array->function_name, "__construct");
+			if (!is_ctor) {
```

Comment:
Could add a test of global functions/closures - I'd guess that the following snippet probably has a misleading error message (not sure what scope->properties_info is)

```php
<?php
function __construct(public $x) {}
```

LGTM otherwise

Reply:
Added a test in https://github.com/php/php-src/pull/5291/commits/717134eff8d82477305d7ca557218c3b66ec9919, error message seems fine. Or did you expect something different there?

Reply:
Ah sorry, I missed that you had `function __construct()` in particular in mind here. Changed the name in https://github.com/php/php-src/pull/5291/commits/7139cdc820a6b1c7e4e2cf276dc909fbe2b4d3c1.

Still works fine though, because `is_ctor` above includes a check that `scope` is not null.

Reply:
Oh, didn't see the `scope` part of the check